applications/GeoMechanicsApplication/tests/cpp_tests/test_geomechanics_erosion_piping.cpp

```diff
@@ -0,0 +1,215 @@
+        // As a result the height curve crosses twice a pipe height growth curve in the search algorithm implemented
+        // in  GeoMechanicsNewtonRaphsonErosionProcessStrategy. The search stops at the second point.
+        for (auto& node : mrModelPart.Nodes()) {
+            node.FastGetSolutionStepValue(WATER_PRESSURE) =
+                pow(node.FastGetSolutionStepValue(WATER_PRESSURE), 0.9725);
+        }
+        return true;
+    }
+
+    void BaseClassFinalizeSolutionStep() override {
+        // to avoid calling the BaseClassFinalizeSolutionStep in tested class.
+    };
+    ModelPart& mrModelPart;
+};
+
+Node::Pointer CreateNodeWithFastSolutionStepValues(
+    ModelPart& rModelPart, int Id, double CoordinateX, double CoordinateY, double WaterPressure)
+{
+    constexpr double coordinate_z = 0.0;
+    auto             p_node = rModelPart.CreateNewNode(Id, CoordinateX, CoordinateY, coordinate_z);
+
+    p_node->FastGetSolutionStepValue(WATER_PRESSURE) = WaterPressure;
+    const array_1d<double, 3> GravitationalAcceleration{0, -9.81, 0};
+    p_node->FastGetSolutionStepValue(VOLUME_ACCELERATION) = GravitationalAcceleration;
+
+    return p_node;
+}
+
+Geometry<Node>::Pointer CreateQuadrilateral2D4N(ModelPart&       rModelPart,
+                                                std::vector<int> NodeIds,
+                                                double           Xmin,
+                                                double           Xmax,
+                                                double           WaterPressureLeft,
+                                                double           WaterPressureRight)
+{
+    constexpr double y_min = -0.1;
+    constexpr double y_max = 0.1;
+
+    return Kratos::make_shared<Quadrilateral2D4<Node>>(
+        CreateNodeWithFastSolutionStepValues(rModelPart, NodeIds[0], Xmin, y_min, WaterPressureLeft),
+        CreateNodeWithFastSolutionStepValues(rModelPart, NodeIds[1], Xmax, y_min, WaterPressureRight),
+        CreateNodeWithFastSolutionStepValues(rModelPart, NodeIds[2], Xmax, y_max, WaterPressureRight),
+        CreateNodeWithFastSolutionStepValues(rModelPart, NodeIds[3], Xmin, y_max, WaterPressureLeft));
+}
+
+KratosExecute::GeoMechanicsNewtonRaphsonErosionProcessStrategyType::Pointer SetupPipingStrategy(Model& rModel)
+{
+    using SparseSpaceType = UblasSpace<double, CompressedMatrix, Vector>;
+    using LocalSpaceType  = UblasSpace<double, Matrix, Vector>;
+
+    auto& r_model_part = rModel.CreateModelPart("ModelPart", 1);
+    r_model_part.AddNodalSolutionStepVariable(WATER_PRESSURE);
+    r_model_part.AddNodalSolutionStepVariable(VOLUME_ACCELERATION);
+    const auto& r_process_info = r_model_part.GetProcessInfo();
+
+    auto p_element_props = r_model_part.CreateNewProperties(0);
+    p_element_props->SetValue(CONSTITUTIVE_LAW, LinearElastic2DInterfaceLaw().Clone());
+
+    auto p_element = make_intrusive<UPwSmallStrainElement<2, 4>>(
```

Comment:
Probably a Pw element is sufficient. The use of the piping elements so far is in a Pw context only.

Reply:
Used Pw element now. This info is used only to create the object. It is not used for testing.